code/modules/mob/mob_defines.dm

```diff
@@ -220,5 +220,5 @@
 	///Override for sound_environments. If this is set the user will always hear a specific type of reverb (Instead of the area defined reverb)
 	var/sound_environment_override = SOUND_ENVIRONMENT_NONE
 
-	///Currently possesses a typing indicator icon
-	var/typing_indicator = FALSE
+	/// The current typing indicator used (usually false)
+	var/mutable_appearance/typing_indicator = FALSE
```

Comment:
make this a global list that stores a mapping of `sprite_used` -> `mutable_appearances` so every mob doesn't have to initialize their own mutable_appearance.

Then just use `GLOB.typing_indicators[sprite_used]` etc